interfaces/prompting/requestprompts/requestprompts.go

```diff
@@ -83,6 +99,59 @@ func (p *Prompt) MarshalJSON() ([]byte, error) {
 	return json.Marshal(toMarshal)
 }
 
+func (p *Prompt) sendReply(outcome prompting.OutcomeType) error {
+	// Reply with any permissions which were previously allowed
+	// If outcome is allow, then reply by allowing all originally-requested
+	// permissions. If outcome is deny, only allow permissions which were
+	// originally requested but have since been allowed by rules, and deny any
+	// remaining permissions.
+	allowedPermission := responseForInterfaceConstraintsOutcome(p.Interface, p.Constraints, outcome)
+	return p.sendReplyWithPermission(allowedPermission)
+}
+
+func responseForInterfaceConstraintsOutcome(iface string, constraints *promptConstraints, outcome prompting.OutcomeType) any {
```

Comment:
Hmm, do we need the `any` return value? I see this is not new but I wonder if we can avoid it.

Reply:
Yes, I have a task to address this later: https://warthogs.atlassian.net/browse/SNAPDENG-30466